### PR TITLE
Stop copying the light probe array to the stack in the shader.

### LINKED ISSUE #11787: Frame unstable in many_foxes example
## Bevy version
main(#11781 )

## \[Optional\] Relevant system information
windows AMD R7-4800H NV 2060

## What you did
Run example "many_foxes"

## What went wrong
fps jump between 90- and 150+

## Additional information

Frame Time show
image

`prepare_windows `will block a long time sometimes
strangely, this phenomenon is intermittent(with an interval about 1 second in my computer)
image

Seems introduced by #10268

### PR DESCRIPTION
**Pull request by pcwalton**
This was causing a severe performance regression when light probes were enabled.

Fixes #11787.

### CODE REVIEW

**Comment by re0312** [comment]
seems also fix #11787

**Comment by NthTensor** [comment]
This really needs to get in for 0.13.

**Review by JMS55 — APPROVED** [review]

**Review by Elabajaba — APPROVED** [review]
Code seems fine. The comment for `is_irradiance_volume` is a bit ugly, but that's just nitpicking.

I'm still seeing a (significantly smaller) 5-10% regression at 4k (GPU timing for the Opaque3d pass, tested on bistro and synty castle scene on my 6800xt with locked clocks), but at 1080p this seems to have brought performance to the same as before irradiance volumes were merged.

File: crates/bevy_pbr/src/light_probe/environment_map.wgsl [hunk 1]
```diff
@@ -30,10 +30,7 @@ fn compute_radiances(
     var radiances: EnvironmentMapRadiances;
 
     // Search for a reflection probe that contains the fragment.
-    var query_result = query_light_probe(
-        light_probes.reflection_probes,
-        light_probes.reflection_probe_count,
-        world_position);
+    var query_result = query_light_probe(world_position, /*is_irradiance_volume=*/ false);
```
Comment (mockersf) [inline crates/bevy_pbr/src/light_probe/environment_map.wgsl:33]: Could you remove it or add a comment on why it's commented out?

**Pull request merged**